backend/cfg/llvmize.ml

```diff
@@ -292,22 +326,132 @@ module F = struct
 
   let ins_branch t label = ins t "br %a" (pp_label t) label
 
-  let ins_alloca t (reg : Reg.t) =
-    ins t "%a = alloca %a" (pp_reg_ident t) reg pp_machtyp_component reg.typ
+  (* [dbg] is used to print the [Reg.t] the allocated identifier corresponds to
+     in the preamble *)
+  let ins_alloca ?dbg t (reg : Reg.t) =
+    let pp_my_dbg ppf () =
+      match dbg with
+      | None -> ()
+      | Some dbg -> pp_dbg_comment ~newline:false ppf dbg Debuginfo.none
+    in
+    ins t "%a = alloca %a %a" (pp_reg_ident t) reg pp_machtyp_component reg.typ
+      pp_my_dbg ()
 
   let ins_branch_cond t cond ifso ifnot =
     ins t "br i1 %a, %a, %a" pp_ident cond (pp_label t) ifso (pp_label t) ifnot
 
+  let ins_branch_cond_ident t cond ifso ifnot =
+    ins t "br i1 %a, label %a, label %a" pp_ident cond pp_ident ifso pp_ident
+      ifnot
+
+  let ins_conv t op ~src ~dst ~src_typ ~dst_typ =
+    ins t "%a = %s %a %a to %a" pp_ident dst op Llvm_typ.pp_t src_typ pp_ident
+      src Llvm_typ.pp_t dst_typ
+
+  let ins_binop t op arg1 arg2 res typ =
+    ins t "%a = %s %a %a, %a" pp_ident res op Llvm_typ.pp_t typ pp_ident arg1
+      pp_ident arg2
+
+  let ins_binop_imm t op arg imm res typ =
+    ins t "%a = %s %a %a, %s" pp_ident res op Llvm_typ.pp_t typ pp_ident arg imm
+
+  let ins_monop t op arg res typ =
+    ins t "%a = %s %a %a" pp_ident res op Llvm_typ.pp_t typ pp_ident arg
+
+  let ins_icmp t cond arg1 arg2 res typ =
+    ins t "%a = icmp %s %a %a, %a" pp_ident res cond Llvm_typ.pp_t typ pp_ident
+      arg1 pp_ident arg2
+
+  let ins_icmp_imm t cond arg imm res typ =
+    ins t "%a = icmp %s %a %a, %s" pp_ident res cond Llvm_typ.pp_t typ pp_ident
+      arg imm
+
+  let ins_fcmp t cond arg1 arg2 res typ =
+    ins t "%a = fcmp %s %a %a, %a" pp_ident res cond Llvm_typ.pp_t typ pp_ident
+      arg1 pp_ident arg2
+
   let load_reg_to_temp t reg =
     let temp = fresh_ident t in
-    ins_load t temp reg;
+    ins_load_reg t temp reg;
     temp
 
+  let load_addr t (addr : Arch.addressing_mode) (i : 'a Cfg.instruction) n =
+    let offset = Arch.addressing_displacement_for_llvmize addr in
+    let arg = load_reg_to_temp t i.arg.(n) in
```

Comment:
Looks like `i` and `n` are only used to in `i.arg.(n)` so the caller can do it:
```suggestion
  let load_addr t (addr : Arch.addressing_mode) (reg : Reg.t) =
    let offset = Arch.addressing_displacement_for_llvmize addr in
    let arg = load_reg_to_temp t reg in
```

Reply:
I guess it was needed to handle more interesting addressing modes.. ok to keep as is, just in case there is some reason to do it again later.